### PR TITLE
refactor: started at items sorting

### PR DESCRIPTION
**Pull request by topliceanurazvan**
## Problem Statement

The **agent-side** sorting logic for test executions was inconsistent and had critical issues that affected the user experience:

1. **Incorrect sorting priority**: Tests and test suites were using EndTime first, then checking if StartTime was "after" EndTime to decide precedence
2. **Agent-cloud inconsistency**: The agent-side sorting logic didn't match the cloud-side implementation, leading to different ordering behaviors
3. **MongoDB aggregation issues**: Agent repositories used simple `$max` operations that didn't properly handle zero time values (`0001-01-01T00:00:00Z`)
4. **Poor user experience**: Lists showed tests sorted by when they ended rather than when they actually started running

## Solution

Implemented **consistent "StartTime with fallback" sorting hierarchy** across all agent-side components:

### 📝 **Sorting Priority Order**
1. **Primary**: `StartTime` / `StartedAt` (when the execution actually began)
2. **Fallback**: `EndTime` / `StatusAt` (when StartTime is null or zero)
3. **Final Fallback**: `Created` time (for items without any executions)

#### API Handler Logic Updates
- **Fixed comparison logic**: Changed from `StartTime.After(EndTime)` to proper zero-time checking with `IsZero()`
- **Consistent fallback**: All handlers now use StartTime as primary with EndTime/StatusAt as fallback
- **Improved comments**: Added clear explanations for the fallback logic

#### MongoDB Repository Updates
- **Enhanced aggregation pipelines**: Replaced simple `$max` with complex `$cond` logic
- **Zero time handling**: Added explicit checks for both `null` and `0001-01-01T00:00:00Z`
- **Consistent patterns**: Applied identical MongoDB logic across all repository functions

### 🌐 **API Handler Updates**
- **`tests.go`** (deprecated v1): Fixed test execution sorting with proper StartTime → EndTime fallback
- **`testsuites.go`** (deprecated v1): Updated test suite execution sorting with consistent logic
- **`testworkflowwithexecutions.go`** (v1): Changed from StatusAt-only to StartedAt → StatusAt → Created hierarchy

### 🗃️ **Repository Layer Updates**
- **`result/mongo/mongo.go`**: Updated 4 aggregation pipelines with proper StartTime prioritization
- **`testresult/mongo/mongo.go`**: Fixed 4 aggregation functions with consistent zero-time handling

### ✅ **Comprehensive Test Coverage**
Added extensive test suites for all API handlers:
- **StartTime precedence verification**
- **EndTime/StatusAt fallback scenarios**
- **Creation time fallback for items without executions**
- **Zero time handling edge cases**
- **Mixed execution scenarios**

### Before (Agent-Side Issues)
```javascript
// Inconsistent and problematic logic
tests: EndTime first, then StartTime.After(EndTime) check     ❌
testsuites: EndTime first, then StartTime.After(EndTime)     ❌  
workflows: StatusAt only                                     ❌
repositories: Simple $max without zero-time handling        ❌
```

### After (Consistent Agent Logic)
```javascript
// Consistent StartTime priority across all components
tests: StartTime → EndTime → Created                        ✅
testsuites: StartTime → EndTime → Created                   ✅
workflows: StartedAt → StatusAt → Created                   ✅
repositories: Complex $cond with zero-time handling         ✅
```

### 🎯 **User Experience**
- **Chronological accuracy**: Agent-side lists now show when tests actually started running
- **Cloud-agent consistency**: Both cloud and agent show identical sorting behavior
- **Predictable behavior**: Most recently started executions appear first consistently

### 🔧 **Technical Benefits**
- **Robust zero-time handling**: Properly handles MongoDB's zero time representation in aggregations
- **Comprehensive test coverage**: All edge cases and scenarios are tested
- **Agent-cloud alignment**: Eliminates discrepancies between cloud and agent sorting

## Repository-Specific Details

This PR specifically addresses the **agent-side implementation** in the testkube repository:

### 📊 **API Layer Changes** (726 additions, 25 deletions)
- 3 API handler files updated with consistent sorting logic
- 3 new test files added with comprehensive scenario coverage

### 🔧 **Repository Layer Changes**
- 8 MongoDB aggregation pipelines updated across 2 repository files
- Complex conditional logic added to handle zero time values properly

### 🧪 **Testing** 
- 618 lines of new test code covering all sorting scenarios
- Tests verify StartTime precedence, fallback behavior, and edge cases

The changes ensure that the agent-side sorting behavior matches the cloud-side implementation, providing a consistent user experience across all Testkube deployments.

### CODE REVIEW

File: pkg/repository/result/mongo/mongo.go [hunk 1]
```diff
@@ -201,7 +201,16 @@ func (r *MongoRepository) slowGetLatestByTest(ctx context.Context, testName stri
 		{"$group": bson.D{
 			{Key: "_id", Value: "$testname"},
 			{Key: "doc", Value: bson.M{"$max": bson.D{
-				{Key: "updatetime", Value: bson.M{"$max": bson.A{"$starttime", "$endtime"}}},
+				{Key: "updatetime", Value: bson.M{"$max": bson.A{
+					bson.M{"$cond": bson.M{
+						"if": bson.M{"$and": bson.A{
+							bson.M{"$ne": bson.A{"$starttime", nil}},
+							bson.M{"$ne": bson.A{"$starttime", bson.M{"$dateFromString": bson.M{"dateString": "0001-01-01T00:00:00Z"}}}},
```
Comment (povilasv) [inline pkg/repository/result/mongo/mongo.go:208]: I am a bit confused about `dateFromString` what are we parsing here?

Comment (topliceanurazvan) [inline pkg/repository/result/mongo/mongo.go:208]: We're parsing Go's zero time representation (`"0001-01-01T00:00:00Z"`) to check if StartTime is actually set. MongoDB stores Go's `time.Time{}` as this string, so we need to convert it back to a date object for proper comparison. This ensures we only use StartTime when it has a real value, otherwise we fall back to EndTime.

Comment (topliceanurazvan) [reply pkg/repository/result/mongo/mongo.go:208]: We're parsing Go's zero time representation (`"0001-01-01T00:00:00Z"`) to check if StartTime is set. MongoDB stores Go's `time.Time{}` as this string, so we need to convert it back to a date object for proper comparison. This ensures we only use StartTime when it has a valid value; otherwise, we fall back to EndTime.

Comment (topliceanurazvan) [reply pkg/repository/result/mongo/mongo.go:208]: The `dateFromString` approach was chosen over simpler alternatives (such as string comparison or null checks) because it provides proper type-safe date comparison and exactly matches Go's `time.IsZero()` semantics in MongoDB aggregation context.

Comment (povilasv) [reply pkg/repository/result/mongo/mongo.go:208]: Makes sense
